ember-core/src/test/scala/org/http4s/ember/core/ParsingSpec.scala

```diff
@@ -254,21 +295,50 @@ class ParsingSpec extends Http4sSuite {
         "Expires: Wed, 21 Oct 2015 07:28:00 GMT\r\n\r\n"
         // "\r\n"
       )
+
     val byteStream: Stream[IO, Byte] = respS
       .flatMap(s =>
         Stream.chunk(Chunk.array(s.getBytes(java.nio.charset.StandardCharsets.US_ASCII))))
 
-    Parser.Response
-      .parser[IO](defaultMaxHeaderLength, None)(byteStream)
-      .flatMap { case (resp, _) =>
-        for {
-          body <- resp.body.through(text.utf8Decode).compile.string
-          trailers <- resp.trailerHeaders
-        } yield (body == "MozillaDeveloperNetwork").&&(
-          trailers.get(Expires).isDefined
-        )
-      }
-      .assert
+    (for {
+      take <- Helpers.taking[IO, Byte](byteStream)
+      result <- Parser.Response
+        .parser[IO](defaultMaxHeaderLength, None)(Array.emptyByteArray, take)
+      body <- result._1.body.through(text.utf8Decode).compile.string
+      trailers <- result._1.trailerHeaders
+    } yield body == "MozillaDeveloperNetwork" && trailers.get(Expires).isDefined).assert
+  }
+
+  test(
+    "Parser.Response.parser should parse a response with Content-Length and return the rest of the stream") {
+    val defaultMaxHeaderLength = 4096
+    val raw =
+      """HTTP/1.1 200 OK
+        |Host: localhost:8080
+        |User-Agent: curl/7.64.1
+        |Accept: */*
```

Comment:
`Host` and `Accept` aren't typical response headers, but they're syntactically valid so I don't think it negates the test.